🌱 Small improvement to the backport policy (#7823)
kubernetes-sigs/cluster-api: CONTRIBUTING.md

#### Backporting a patch

We generally do not accept PRs directly against release branches, while we might accept backports of fixes/changes already
merged into the main branch.

Any backport MUST not be breaking for either API or behavioral changes. 

> We generally allow backport of following changes to all the supported branches:

Reviewer: ```suggestion
We generally allow backports of following changes to all supported branches:
```

> - Critical bugs fix, security issues fix, or fix for bugs without easy workarounds.

Reviewer: ```suggestion
- Critical bug fixes, security issue fixes, or fixes for bugs without easy workarounds.
```

- Dependency bumps for CVE (usually limited to CVE resolution; backports of non-CVE related version bumps are considered exceptions to be evaluated case by case)

> - Cert-manager version bump (to avoid having branch using cert-manager versions out of support, when possible)

Reviewer: ```suggestion
- Cert-manager version bumps (to avoid having releases with cert-manager versions that are out of support, when possible)
```
nit: sorry I know it was like this before. Just while we're already here

- Changes required to support new Kubernetes versions, when possible. See [supported Kubernetes versions](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions) for more details.

> We generally allow backport of following changes only to the latest supported branch:

Reviewer: ```suggestion
We generally allow backports of following changes only to the latest supported branch:
```

> - Improvements to existing docs (the latest stable version hosts the current version of the book)

Reviewer: ```suggestion
- Improvements to existing docs (the latest supported branch hosts the current version of the book)
```
Just to keep it in sync with how we call it one line above

#### Backporting a patch

> We generally do not accept PRs against older release branches, while we might accept backport of fix/changes already
> applied to the main branch.

Reviewer: ```suggestion
We generally do not accept PRs directly against release branches, while we might accept backports of fixes/changes already
merged into the main branch.
```

We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
backport MUST not be breaking for either API or behavioral changes. 

> In order to keep up the project up-to-date with the ecosystem, maintainers can choose to accept backports of:

Reviewer: ```suggestion
In order to keep the project up-to-date with the ecosystem, maintainers can choose to accept backports of:
```

> - Dependency bump (usually limited to CVE resolution; backport of non-CVE related version bump are considered exceptions to be evaluated case by case)

Reviewer 2: ```suggestion
- Dependency bumps (usually limited to CVE resolution; backports of non-CVE related version bumps are considered exceptions to be evaluated case by case)
```

- Cert-manager bump (to avoid having branch using cert-manager versions out of support, when possible)

> - Changes required to support new Kubernetes versions, when possible. see [supported Kubernetes versions](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions) for more details.

Reviewer 3: nit:
```suggestion
- Changes required to support new Kubernetes versions, when possible. See [supported Kubernetes versions](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions) for more details.
```

Additionally, in order to improve user/developer experience maintainers can choose to accept backports of:

> - Doc improvements (usually limited to latest stable version which hosts the current version of the book; backport to older releases are considered exceptions to be evaluated case by case)

Reviewer 2: ```suggestion
- Doc improvements (usually limited to latest stable version which hosts the current version of the book; backports to older releases are considered exceptions to be evaluated case by case)
```

> - Improvements to CI signal (usually limited to latest stable version; backport to older releases are considered exceptions to be evaluated case by case)

Reviewer 2: ```suggestion
- Improvements to CI signal (usually limited to latest stable version; backports to older releases are considered exceptions to be evaluated case by case)
```

> - Improvements to the test framework (usually limited to latest stable version; backport to older releases are considered exceptions to be evaluated case by case)

Reviewer 2: ```suggestion
- Improvements to the test framework (usually limited to latest stable version; backports to older releases are considered exceptions to be evaluated case by case)
```

Like any other activity in the project, backporting a fix/change also requires that someone volunteers for the task.
The cherry pick bot can be used for automating some of this work thus reducing the effort for this job. But

> independent of that, the community is expected to also help in taking care of older Cluster API release branches while still

Reviewer 4: > the community is expected to also help in taking care of older Cluster API release branches 

not sure this sentence is necessary, it's a little vague

What do you think of this instead?

> Like any other activity in the project, backporting a fix/change is a community-driven effort and requires that someone volunteers to own the task. In most cases, the [cherry-pick bot](https://pkg.go.dev/github.com/k8s-ci-robot/test-infra/prow/external-plugins/cherrypicker) can (and should) be used to automate opening a PR.

supported (currently n and n-1).

> We generally do not accept backports to Cluster API release branches that are out of support.

Reviewer 4: ```suggestion
We generally do not accept backports to Cluster API release branches that are [out of support](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#support-and-guarantees).
```

> Like any other activity in the project, also backporting a fix/change requires that someone volunteers for the task;
> the cherry pick bot could be used for automating some of this work thus reducting the effort for this job, but,
> no matter of that, the community is expected to help also in taking care of older Cluster API release branches while still
> supported (currently n and n-1).

Reviewer: ```suggestion
Like any other activity in the project, backporting a fix/change also requires that someone volunteers for the task.
The cherry pick bot can be used for automating some of this work thus reducing the effort for this job. But
independent of that, the community is expected to also help in taking care of older Cluster API release branches while still
supported (currently n and n-1).
```

> We generally do not accept backport to Cluster API release branches out of support.

Reviewer 2: ```suggestion
We generally do not accept backports to Cluster API release branches that are out of support.
```
